// @nakarmi23/bikram-sambat/dist/index.js
"use strict";
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  // If the importer is in node compatibility mode or this is not an ESM
  // file that has been converted to a CommonJS file using a Babel-
  // compatible transform (i.e. "__esModule" has not been set), then set
  // "default" to the CommonJS "module.exports" for node compatibility.
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/index.ts
var src_exports = {};
__export(src_exports, {
  BikramSambat: () => BikramSambat,
  default: () => src_default
});
module.exports = __toCommonJS(src_exports);
var import_dayjs2 = __toESM(require("dayjs"));

// src/constraints/nepali-english-calender-maps.ts
var nepEngCalenderMaps = [
  {
    nepYear: 1970,
    startDate: "1913-04-13T00:00:00",
    endDate: "1914-04-12T00:00:00"
  },
  {
    nepYear: 1971,
    startDate: "1914-04-13T00:00:00",
    endDate: "1915-04-12T00:00:00"
  },
  {
    nepYear: 1972,
    startDate: "1915-04-13T00:00:00",
    endDate: "1916-04-11T00:00:00"
  },
  {
    nepYear: 1973,
    startDate: "1916-04-12T00:00:00",
    endDate: "1917-04-11T00:00:00"
  },
  {
    nepYear: 1974,
    startDate: "1917-04-12T00:00:00",
    endDate: "1918-04-11T00:00:00"
  },
  {
    nepYear: 1975,
    startDate: "1918-04-12T00:00:00",
    endDate: "1919-04-11T00:00:00"
  },
  {
    nepYear: 1976,
    startDate: "1919-04-12T00:00:00",
    endDate: "1920-04-11T00:00:00"
  },
  {
    nepYear: 1977,
    startDate: "1920-04-12T00:00:00",
    endDate: "1921-04-11T00:00:00"
  },
  {
    nepYear: 1978,
    startDate: "1921-04-12T00:00:00",
    endDate: "1922-04-11T00:00:00"
  },
  {
    nepYear: 1979,
    startDate: "1922-04-12T00:00:00",
    endDate: "1923-04-11T00:00:00"
  },
  {
    nepYear: 1980,
    startDate: "1923-04-12T00:00:00",
    endDate: "1924-04-11T00:00:00"
  },
  {
    nepYear: 1981,
    startDate: "1924-04-12T00:00:00",
    endDate: "1925-04-11T00:00:00"
  },
  {
    nepYear: 1982,
    startDate: "1925-04-12T00:00:00",
    endDate: "1926-04-11T00:00:00"
  },
  {
    nepYear: 1983,
    startDate: "1926-04-12T00:00:00",
    endDate: "1927-04-11T00:00:00"
  },
  {
    nepYear: 1984,
    startDate: "1927-04-12T00:00:00",
    endDate: "1928-04-11T00:00:00"
  },
  {
    nepYear: 1985,
    startDate: "1928-04-12T00:00:00",
    endDate: "1929-04-11T00:00:00"
  },
  {
    nepYear: 1986,
    startDate: "1929-04-12T00:00:00",
    endDate: "1930-04-11T00:00:00"
  },
  {
    nepYear: 1987,
    startDate: "1930-04-12T00:00:00",
    endDate: "1931-04-12T00:00:00"
  },
  {
    nepYear: 1988,
    startDate: "1931-04-13T00:00:00",
    endDate: "1932-04-12T00:00:00"
  },
  {
    nepYear: 1989,
    startDate: "1932-04-13T00:00:00",
    endDate: "1933-04-12T00:00:00"
  },
  {
    nepYear: 1990,
    startDate: "1933-04-13T00:00:00",
    endDate: "1934-04-12T00:00:00"
  },
  {
    nepYear: 1991,
    startDate: "1934-04-13T00:00:00",
    endDate: "1935-04-12T00:00:00"
  },
  {
    nepYear: 1992,
    startDate: "1935-04-13T00:00:00",
    endDate: "1936-04-12T00:00:00"
  },
  {
    nepYear: 1993,
    startDate: "1936-04-13T00:00:00",
    endDate: "1937-04-12T00:00:00"
  },
  {
    nepYear: 1994,
    startDate: "1937-04-13T00:00:00",
    endDate: "1938-04-12T00:00:00"
  },
  {
    nepYear: 1995,
    startDate: "1938-04-13T00:00:00",
    endDate: "1939-04-12T00:00:00"
  },
  {
    nepYear: 1996,
    startDate: "1939-04-13T00:00:00",
    endDate: "1940-04-12T00:00:00"
  },
  {
    nepYear: 1997,
    startDate: "1940-04-13T00:00:00",
    endDate: "1941-04-12T00:00:00"
  },
  {
    nepYear: 1998,
    startDate: "1941-04-13T00:00:00",
    endDate: "1942-04-12T00:00:00"
  },
  {
    nepYear: 1999,
    startDate: "1942-04-13T00:00:00",
    endDate: "1943-04-13T00:00:00"
  },
  {
    nepYear: 2e3,
    startDate: "1943-04-14T00:00:00",
    endDate: "1944-04-12T00:00:00"
  },
  {
    nepYear: 2001,
    startDate: "1944-04-13T00:00:00",
    endDate: "1945-04-12T00:00:00"
  },
  {
    nepYear: 2002,
    startDate: "1945-04-13T00:00:00",
    endDate: "1946-04-12T00:00:00"
  },
  {
    nepYear: 2003,
    startDate: "1946-04-13T00:00:00",
    endDate: "1947-04-13T00:00:00"
  },
  {
    nepYear: 2004,
    startDate: "1947-04-14T00:00:00",
    endDate: "1948-04-12T00:00:00"
  },
  {
    nepYear: 2005,
    startDate: "1948-04-13T00:00:00",
    endDate: "1949-04-12T00:00:00"
  },
  {
    nepYear: 2006,
    startDate: "1949-04-13T00:00:00",
    endDate: "1950-04-12T00:00:00"
  },
  {
    nepYear: 2007,
    startDate: "1950-04-13T00:00:00",
    endDate: "1951-04-13T00:00:00"
  },
  {
    nepYear: 2008,
    startDate: "1951-04-14T00:00:00",
    endDate: "1952-04-12T00:00:00"
  },
  {
    nepYear: 2009,
    startDate: "1952-04-13T00:00:00",
    endDate: "1953-04-12T00:00:00"
  },
  {
    nepYear: 2010,
    startDate: "1953-04-13T00:00:00",
    endDate: "1954-04-12T00:00:00"
  },
  {
    nepYear: 2011,
    startDate: "1954-04-13T00:00:00",
    endDate: "1955-04-13T00:00:00"
  },
  {
    nepYear: 2012,
    startDate: "1955-04-14T00:00:00",
    endDate: "1956-04-12T00:00:00"
  },
  {
    nepYear: 2013,
    startDate: "1956-04-13T00:00:00",
    endDate: "1957-04-12T00:00:00"
  },
  {
    nepYear: 2014,
    startDate: "1957-04-13T00:00:00",
    endDate: "1958-04-12T00:00:00"
  },
  {
    nepYear: 2015,
    startDate: "1958-04-13T00:00:00",
    endDate: "1959-04-13T00:00:00"
  },
  {
    nepYear: 2016,
    startDate: "1959-04-14T00:00:00",
    endDate: "1960-04-12T00:00:00"
  },
  {
    nepYear: 2017,
    startDate: "1960-04-13T00:00:00",
    endDate: "1961-04-12T00:00:00"
  },
  {
    nepYear: 2018,
    startDate: "1961-04-13T00:00:00",
    endDate: "1962-04-12T00:00:00"
  },
  {
    nepYear: 2019,
    startDate: "1962-04-13T00:00:00",
    endDate: "1963-04-13T00:00:00"
  },
  {
    nepYear: 2020,
    startDate: "1963-04-14T00:00:00",
    endDate: "1964-04-12T00:00:00"
  },
  {
    nepYear: 2021,
    startDate: "1964-04-13T00:00:00",
    endDate: "1965-04-12T00:00:00"
  },
  {
    nepYear: 2022,
    startDate: "1965-04-13T00:00:00",
    endDate: "1966-04-12T00:00:00"
  },
  {
    nepYear: 2023,
    startDate: "1966-04-13T00:00:00",
    endDate: "1967-04-13T00:00:00"
  },
  {
    nepYear: 2024,
    startDate: "1967-04-14T00:00:00",
    endDate: "1968-04-12T00:00:00"
  },
  {
    nepYear: 2025,
    startDate: "1968-04-13T00:00:00",
    endDate: "1969-04-12T00:00:00"
  },
  {
    nepYear: 2026,
    startDate: "1969-04-13T00:00:00",
    endDate: "1970-04-13T00:00:00"
  },
  {
    nepYear: 2027,
    startDate: "1970-04-14T00:00:00",
    endDate: "1971-04-13T00:00:00"
  },
  {
    nepYear: 2028,
    startDate: "1971-04-14T00:00:00",
    endDate: "1972-04-12T00:00:00"
  },
  {
    nepYear: 2029,
    startDate: "1972-04-13T00:00:00",
    endDate: "1973-04-12T00:00:00"
  },
  {
    nepYear: 2030,
    startDate: "1973-04-13T00:00:00",
    endDate: "1974-04-13T00:00:00"
  },
  {
    nepYear: 2031,
    startDate: "1974-04-14T00:00:00",
    endDate: "1975-04-13T00:00:00"
  },
  {
    nepYear: 2032,
    startDate: "1975-04-14T00:00:00",
    endDate: "1976-04-12T00:00:00"
  },
  {
    nepYear: 2033,
    startDate: "1976-04-13T00:00:00",
    endDate: "1977-04-12T00:00:00"
  },
  {
    nepYear: 2034,
    startDate: "1977-04-13T00:00:00",
    endDate: "1978-04-13T00:00:00"
  },
  {
    nepYear: 2035,
    startDate: "1978-04-14T00:00:00",
    endDate: "1979-04-13T00:00:00"
  },
  {
    nepYear: 2036,
    startDate: "1979-04-14T00:00:00",
    endDate: "1980-04-12T00:00:00"
  },
  {
    nepYear: 2037,
    startDate: "1980-04-13T00:00:00",
    endDate: "1981-04-12T00:00:00"
  },
  {
    nepYear: 2038,
    startDate: "1981-04-13T00:00:00",
    endDate: "1982-04-13T00:00:00"
  },
  {
    nepYear: 2039,
    startDate: "1982-04-14T00:00:00",
    endDate: "1983-04-13T00:00:00"
  },
  {
    nepYear: 2040,
    startDate: "1983-04-14T00:00:00",
    endDate: "1984-04-12T00:00:00"
  },
  {
    nepYear: 2041,
    startDate: "1984-04-13T00:00:00",
    endDate: "1985-04-12T00:00:00"
  },
  {
    nepYear: 2042,
    startDate: "1985-04-13T00:00:00",
    endDate: "1986-04-13T00:00:00"
  },
  {
    nepYear: 2043,
    startDate: "1986-04-14T00:00:00",
    endDate: "1987-04-13T00:00:00"
  },
  {
    nepYear: 2044,
    startDate: "1987-04-14T00:00:00",
    endDate: "1988-04-12T00:00:00"
  },
  {
    nepYear: 2045,
    startDate: "1988-04-13T00:00:00",
    endDate: "1989-04-12T00:00:00"
  },
  {
    nepYear: 2046,
    startDate: "1989-04-13T00:00:00",
    endDate: "1990-04-13T00:00:00"
  },
  {
    nepYear: 2047,
    startDate: "1990-04-14T00:00:00",
    endDate: "1991-04-13T00:00:00"
  },
  {
    nepYear: 2048,
    startDate: "1991-04-14T00:00:00",
    endDate: "1992-04-12T00:00:00"
  },
  {
    nepYear: 2049,
    startDate: "1992-04-13T00:00:00",
    endDate: "1993-04-12T00:00:00"
  },
  {
    nepYear: 2050,
    startDate: "1993-04-13T00:00:00",
    endDate: "1994-04-13T00:00:00"
  },
  {
    nepYear: 2051,
    startDate: "1994-04-14T00:00:00",
    endDate: "1995-04-13T00:00:00"
  },
  {
    nepYear: 2052,
    startDate: "1995-04-14T00:00:00",
    endDate: "1996-04-12T00:00:00"
  },
  {
    nepYear: 2053,
    startDate: "1996-04-13T00:00:00",
    endDate: "1997-04-12T00:00:00"
  },
  {
    nepYear: 2054,
    startDate: "1997-04-13T00:00:00",
    endDate: "1998-04-13T00:00:00"
  },
  {
    nepYear: 2055,
    startDate: "1998-04-14T00:00:00",
    endDate: "1999-04-13T00:00:00"
  },
  {
    nepYear: 2056,
    startDate: "1999-04-14T00:00:00",
    endDate: "2000-04-12T00:00:00"
  },
  {
    nepYear: 2057,
    startDate: "2000-04-13T00:00:00",
    endDate: "2001-04-13T00:00:00"
  },
  {
    nepYear: 2058,
    startDate: "2001-04-14T00:00:00",
    endDate: "2002-04-13T00:00:00"
  },
  {
    nepYear: 2059,
    startDate: "2002-04-14T00:00:00",
    endDate: "2003-04-13T00:00:00"
  },
  {
    nepYear: 2060,
    startDate: "2003-04-14T00:00:00",
    endDate: "2004-04-12T00:00:00"
  },
  {
    nepYear: 2061,
    startDate: "2004-04-13T00:00:00",
    endDate: "2005-04-13T00:00:00"
  },
  {
    nepYear: 2062,
    startDate: "2005-04-14T00:00:00",
    endDate: "2006-04-13T00:00:00"
  },
  {
    nepYear: 2063,
    startDate: "2006-04-14T00:00:00",
    endDate: "2007-04-13T00:00:00"
  },
  {
    nepYear: 2064,
    startDate: "2007-04-14T00:00:00",
    endDate: "2008-04-12T00:00:00"
  },
  {
    nepYear: 2065,
    startDate: "2008-04-13T00:00:00",
    endDate: "2009-04-13T00:00:00"
  },
  {
    nepYear: 2066,
    startDate: "2009-04-14T00:00:00",
    endDate: "2010-04-13T00:00:00"
  },
  {
    nepYear: 2067,
    startDate: "2010-04-14T00:00:00",
    endDate: "2011-04-13T00:00:00"
  },
  {
    nepYear: 2068,
    startDate: "2011-04-14T00:00:00",
    endDate: "2012-04-12T00:00:00"
  },
  {
    nepYear: 2069,
    startDate: "2012-04-13T00:00:00",
    endDate: "2013-04-13T00:00:00"
  },
  {
    nepYear: 2070,
    startDate: "2013-04-14T00:00:00",
    endDate: "2014-04-13T00:00:00"
  },
  {
    nepYear: 2071,
    startDate: "2014-04-14T00:00:00",
    endDate: "2015-04-13T00:00:00"
  },
  {
    nepYear: 2072,
    startDate: "2015-04-14T00:00:00",
    endDate: "2016-04-12T00:00:00"
  },
  {
    nepYear: 2073,
    startDate: "2016-04-13T00:00:00",
    endDate: "2017-04-13T00:00:00"
  },
  {
    nepYear: 2074,
    startDate: "2017-04-14T00:00:00",
    endDate: "2018-04-13T00:00:00"
  },
  {
    nepYear: 2075,
    startDate: "2018-04-14T00:00:00",
    endDate: "2019-04-13T00:00:00"
  },
  {
    nepYear: 2076,
    startDate: "2019-04-14T00:00:00",
    endDate: "2020-04-12T00:00:00"
  },
  {
    nepYear: 2077,
    startDate: "2020-04-13T00:00:00",
    endDate: "2021-04-13T00:00:00"
  },
  {
    nepYear: 2078,
    startDate: "2021-04-14T00:00:00",
    endDate: "2022-04-13T00:00:00"
  },
  {
    nepYear: 2079,
    startDate: "2022-04-14T00:00:00",
    endDate: "2023-04-13T00:00:00"
  },
  {
    nepYear: 2080,
    startDate: "2023-04-14T00:00:00",
    endDate: "2024-04-12T00:00:00"
  },
  {
    nepYear: 2081,
    startDate: "2024-04-13T00:00:00",
    endDate: "2025-04-13T00:00:00"
  },
  {
    nepYear: 2082,
    startDate: "2025-04-14T00:00:00",
    endDate: "2026-04-13T00:00:00"
  },
  {
    nepYear: 2083,
    startDate: "2026-04-14T00:00:00",
    endDate: "2027-04-13T00:00:00"
  },
  {
    nepYear: 2084,
    startDate: "2027-04-14T00:00:00",
    endDate: "2028-04-12T00:00:00"
  },
  {
    nepYear: 2085,
    startDate: "2028-04-13T00:00:00",
    endDate: "2029-04-13T00:00:00"
  },
  {
    nepYear: 2086,
    startDate: "2029-04-14T00:00:00",
    endDate: "2030-04-13T00:00:00"
  },
  {
    nepYear: 2087,
    startDate: "2030-04-14T00:00:00",
    endDate: "2031-04-14T00:00:00"
  },
  {
    nepYear: 2088,
    startDate: "2031-04-15T00:00:00",
    endDate: "2032-04-13T00:00:00"
  },
  {
    nepYear: 2089,
    startDate: "2032-04-14T00:00:00",
    endDate: "2033-04-13T00:00:00"
  },
  {
    nepYear: 2090,
    startDate: "2033-04-14T00:00:00",
    endDate: "2034-04-13T00:00:00"
  },
  {
    nepYear: 2091,
    startDate: "2034-04-14T00:00:00",
    endDate: "2035-04-13T00:00:00"
  },
  {
    nepYear: 2092,
    startDate: "2035-04-14T00:00:00",
    endDate: "2036-04-13T00:00:00"
  },
  {
    nepYear: 2093,
    startDate: "2036-04-14T00:00:00",
    endDate: "2037-04-13T00:00:00"
  },
  {
    nepYear: 2094,
    startDate: "2037-04-14T00:00:00",
    endDate: "2038-04-13T00:00:00"
  },
  {
    nepYear: 2095,
    startDate: "2038-04-14T00:00:00",
    endDate: "2039-04-13T00:00:00"
  },
  {
    nepYear: 2096,
    startDate: "2039-04-14T00:00:00",
    endDate: "2040-04-13T00:00:00"
  },
  {
    nepYear: 2097,
    startDate: "2040-04-14T00:00:00",
    endDate: "2041-04-13T00:00:00"
  },
  {
    nepYear: 2098,
    startDate: "2041-04-14T00:00:00",
    endDate: "2042-04-13T00:00:00"
  },
  {
    nepYear: 2099,
    startDate: "2042-04-14T00:00:00",
    endDate: "2043-04-13T00:00:00"
  },
  {
    nepYear: 2100,
    startDate: "2043-04-14T00:00:00",
    endDate: "2044-04-13T00:00:00"
  },
  {
    nepYear: 2101,
    startDate: "2044-04-14T00:00:00",
    endDate: "2045-04-13T00:00:00"
  },
  {
    nepYear: 2102,
    startDate: "2045-04-14T00:00:00",
    endDate: "2046-04-13T00:00:00"
  },
  {
    nepYear: 2103,
    startDate: "2046-04-14T00:00:00",
    endDate: "2047-04-13T00:00:00"
  },
  {
    nepYear: 2104,
    startDate: "2047-04-14T00:00:00",
    endDate: "2048-04-13T00:00:00"
  },
  {
    nepYear: 2105,
    startDate: "2048-04-14T00:00:00",
    endDate: "2049-04-14T00:00:00"
  },
  {
    nepYear: 2106,
    startDate: "2049-04-15T00:00:00",
    endDate: "2050-04-14T00:00:00"
  },
  {
    nepYear: 2107,
    startDate: "2050-04-15T00:00:00",
    endDate: "2051-04-14T00:00:00"
  },
  {
    nepYear: 2108,
    startDate: "2051-04-15T00:00:00",
    endDate: "2052-04-14T00:00:00"
  },
  {
    nepYear: 2109,
    startDate: "2052-04-15T00:00:00",
    endDate: "2053-04-14T00:00:00"
  },
  {
    nepYear: 2110,
    startDate: "2053-04-15T00:00:00",
    endDate: "2054-04-14T00:00:00"
  },
  {
    nepYear: 2111,
    startDate: "2054-04-15T00:00:00",
    endDate: "2055-04-14T00:00:00"
  }
];

// src/constraints/nepali-english-month-name.ts
var months = [
  { id: 1, engName: "January", nepName: "Baishakh" },
  { id: 2, engName: "February", nepName: "Jestha" },
  { id: 3, engName: "March", nepName: "Ashadh" },
  { id: 4, engName: "April", nepName: "Shrawan" },
  { id: 5, engName: "May", nepName: "Bhadra" },
  { id: 6, engName: "June", nepName: "Ashwin" },
  { id: 7, engName: "July", nepName: "Kartik" },
  { id: 8, engName: "August", nepName: "Mangsir" },
  { id: 9, engName: "September", nepName: "Poush" },
  { id: 10, engName: "October", nepName: "Magh" },
  { id: 11, engName: "November", nepName: "Falgun" },
  { id: 12, engName: "December", nepName: "Chaitra" }
];

// src/constraints/nepali-month-total-days.ts
var nepDateNoOfDays = [
  {
    m1: 31,
    m2: 31,
    m3: 32,
    m4: 31,
    m5: 31,
    m6: 31,
    m7: 30,
    m8: 29,
    m9: 30,
    m10: 29,
    m11: 30,
    m12: 30,
    nepYear: 1970
  },
  {
    m1: 31,
    m2: 31,
    m3: 32,
    m4: 31,
    m5: 32,
    m6: 30,
    m7: 30,
    m8: 29,
    m9: 30,
    m10: 29,
    m11: 30,
    m12: 30,
    nepYear: 1971
  },
  {
    m1: 31,
    m2: 32,
    m3: 31,
    m4: 32,
    m5: 31,
    m6: 30,
    m7: 30,
    m8: 30,
    m9: 29,
    m10: 29,
    m11: 30,
    m12: 30,
    nepYear: 1972
  },
  {
    m1: 30,
    m2: 32,
    m3: 31,
    m4: 32,
    m5: 31,
    m6: 30,
    m7: 30,
    m8: 30,
    m9: 29,
    m10: 30,
    m11: 29,
    m12: 31,
    nepYear: 1973
  },
  {
    m1: 31,
    m2: 31,
    m3: 32,
    m4: 31,
    m5: 31,
    m6: 31,
    m7: 30,
    m8: 29,
    m9: 30,
    m10: 29,
    m11: 30,
    m12: 30,
    nepYear: 1974
  },
  {
    m1: 31,
    m2: 31,
    m3: 32,
    m4: 32,
    m5: 31,
    m6: 30,
    m7: 30,
    m8: 29,
    m9: 30,
    m10: 29,
    m11: 30,
    m12: 30,
    nepYear: 1975
  },
  {
    m1: 31,
    m2: 32,
    m3: 31,
    m4: 32,
    m5: 31,
    m6: 30,
    m7: 30,
    m8: 30,
    m9: 29,
    m10: 29,
    m11: 30,
    m12: 31,
    nepYear: 1976
  },
  {
    m1: 30,
    m2: 32,
    m3: 31,
    m4: 32,
    m5: 31,
    m6: 31,
    m7: 29,
    m8: 30,
    m9: 29,
    m10: 30,
    m11: 29,
    m12: 31,
    nepYear: 1977
  },
  {
    m1: 31,
    m2: 31,
    m3: 32,
    m4: 31,
    m5: 31,
    m6: 31,
    m7: 30,
    m8: 29,
    m9: 30,
    m10: 29,
    m11: 30,
    m12: 30,
    nepYear: 1978
  },
  {
    m1: 31,
    m2: 31,
    m3: 32,
    m4: 32,
    m5: 31,
    m6: 30,
    m7: 30,
    m8: 29,
    m9: 30,
    m10: 29,
    m11: 30,
    m12: 30,
    nepYear: 1979
  },
  {
    m1: 31,
    m2: 32,
    m3: 31,
    m4: 32,
    m5: 31,
    m6: 30,
    m7: 30,
    m8: 30,
    m9: 29,
    m10: 29,
    m11: 30,
    m12: 31,
    nepYear: 1980
  },
  {
    m1: 31,
    m2: 31,
    m3: 31,
    m4: 32,
    m5: 31,
    m6: 31,
    m7: 29,
    m8: 30,
    m9: 30,
    m10: 29,
    m11: 30,
    m12: 30,
    nepYear: 1981
  },
  {
    m1: 31,
    m2: 31,
    m3: 32,
    m4: 31,
    m5: 31,
    m6: 31,
    m7: 30,
    m8: 29,
    m9: 30,
    m10: 29,
    m11: 30,
    m12: 30,
    nepYear: 1982
  },
  {
    m1: 31,
    m2: 31,
    m3: 32,
    m4: 32,
    m5: 31,
    m6: 30,
    m7: 30,
    m8: 29,
    m9: 30,
    m10: 29,
    m11: 30,
    m12: 30,
    nepYear: 1983
  },
  {
    m1: 31,
    m2: 32,
    m3: 31,
    m4: 32,
    m5: 31,
    m6: 30,
    m7: 30,
    m8: 30,
    m9: 29,
    m10: 29,
    m11: 30,
    m12: 31,
    nepYear: 1984
  },
  {
    m1: 31,
    m2: 31,
    m3: 31,
    m4: 32,
    m5: 31,
    m6: 31,
    m7: 29,
    m8: 30,
    m9: 30,
    m10: 29,
    m11: 30,
    m12: 30,
    nepYear: 1985
  },
  {
    m1: 31,
    m2: 31,
    m3: 32,
    m4: 31,
    m5: 31,
    m6: 31,
    m7: 30,
    m8: 29,
    m9: 30,
    m10: 29,
    m11: 30,
    m12: 30,
    nepYear: 1986
  },
  {
    m1: 31,
    m2: 32,
    m3: 32,
    m4: 32,
    m5: 31,
    m6: 30,
    m7: 30,
    m8: 29,
    m9: 30,
    m10: 29,
    m11: 30,
    m12: 30,
    nepYear: 1987
  },
  {
    m1: 31,
    m2: 32,
    m3: 31,
    m4: 32,
    m5: 31,
    m6: 30,
    m7: 30,
    m8: 30,
    m9: 29,
    m10: 29,
    m11: 30,
    m12: 31,
    nepYear: 1988
  },
  {
    m1: 31,
    m2: 31,
    m3: 31,
    m4: 32,
    m5: 31,
    m6: 31,
    m7: 30,
    m8: 29,
    m9: 30,
    m10: 29,
    m11: 30,
    m12: 30,
    nepYear: 1989
  },
  {
    m1: 31,
    m2: 31,
    m3: 32,
    m4: 31,
    m5: 31,
    m6: 31,
    m7: 30,
    m8: 29,
    m9: 30,
    m10: 29,
    m11: 30,
    m12: 30,
    nepYear: 1990
  },
  {
    m1: 31,
    m2: 32,
    m3: 31,
    m4: 32,
    m5: 31,
    m6: 30,
    m7: 30,
    m8: 30,
    m9: 29,
    m10: 29,
    m11: 30,
    m12: 30,
    nepYear: 1991
  },
  {
    m1: 31,
    m2: 32,
    m3: 31,
    m4: 32,
    m5: 31,
    m6: 30,
    m7: 30,
    m8: 30,
    m9: 29,
    m10: 30,
    m11: 29,
    m12: 31,
    nepYear: 1992
  },
  {
    m1: 31,
    m2: 31,
    m3: 31,
    m4: 32,
    m5: 31,
    m6: 31,
    m7: 30,
    m8: 29,
    m9: 30,
    m10: 29,
    m11: 30,
    m12: 30,
    nepYear: 1993
  },
  {
    m1: 31,
    m2: 31,
    m3: 32,
    m4: 31,
    m5: 31,
    m6: 31,
    m7: 30,
    m8: 29,
    m9: 30,
    m10: 29,
    m11: 30,
    m12: 30,
    nepYear: 1994
  },
  {
    m1: 31,
    m2: 32,
    m3: 31,
    m4: 32,
    m5: 31,
    m6: 30,
    m7: 30,
    m8: 30,
    m9: 29,
    m10: 29,
    m11: 30,
    m12: 30,
    nepYear: 1995
  },
  {
    m1: 31,
    m2: 32,
    m3: 31,
    m4: 32,
    m5: 31,
    m6: 30,
    m7: 30,
    m8: 30,
    m9: 29,
    m10: 30,
    m11: 29,
    m12: 31,
    nepYear: 1996
  },
  {
    m1: 31,
    m2: 31,
    m3: 32,
    m4: 31,
    m5: 31,
    m6: 31,
    m7: 30,
    m8: 29,
    m9: 30,
    m10: 29,
    m11: 30,
    m12: 30,
    nepYear: 1997
  },
  {
    m1: 31,
    m2: 31,
    m3: 32,
    m4: 31,
    m5: 31,
    m6: 31,
    m7: 30,
    m8: 29,
    m9: 30,
    m10: 29,
    m11: 30,
    m12: 30,
    nepYear: 1998
  },
  {
    m1: 31,
    m2: 32,
    m3: 31,
    m4: 32,
    m5: 31,
    m6: 30,
    m7: 30,
    m8: 30,
    m9: 29,
    m10: 29,
    m11: 30,
    m12: 31,
    nepYear: 1999
  },
  {
    m1: 30,
    m2: 32,
    m3: 31,
    m4: 32,
    m5: 31,
    m6: 30,
    m7: 30,
    m8: 30,
    m9: 29,
    m10: 30,
    m11: 29,
    m12: 31,
    nepYear: 2e3
  },
  {
    m1: 31,
    m2: 31,
    m3: 32,
    m4: 31,
    m5: 31,
    m6: 31,
    m7: 30,
    m8: 29,
    m9: 30,
    m10: 29,
    m11: 30,
    m12: 30,
    nepYear: 2001
  },
  {
    m1: 31,
    m2: 31,
    m3: 32,
    m4: 32,
    m5: 31,
    m6: 30,
    m7: 30,
    m8: 29,
    m9: 30,
    m10: 29,
    m11: 30,
    m12: 30,
    nepYear: 2002
  },
  {
    m1: 31,
    m2: 32,
    m3: 31,
    m4: 32,
    m5: 31,
    m6: 30,
    m7: 30,
    m8: 30,
    m9: 29,
    m10: 29,
    m11: 30,
    m12: 31,
    nepYear: 2003
  },
  {
    m1: 30,
    m2: 32,
    m3: 31,
    m4: 32,
    m5: 31,
    m6: 30,
    m7: 30,
    m8: 30,
    m9: 29,
    m10: 30,
    m11: 29,
    m12: 31,
    nepYear: 2004
  },
  {
    m1: 31,
    m2: 31,
    m3: 32,
    m4: 31,
    m5: 31,
    m6: 31,
    m7: 30,
    m8: 29,
    m9: 30,
    m10: 29,
    m11: 30,
    m12: 30,
    nepYear: 2005
  },
  {
    m1: 31,
    m2: 31,
    m3: 32,
    m4: 32,
    m5: 31,
    m6: 30,
    m7: 30,
    m8: 29,
    m9: 30,
    m10: 29,
    m11: 30,
    m12: 30,
    nepYear: 2006
  },
  {
    m1: 31,
    m2: 32,
    m3: 31,
    m4: 32,
    m5: 31,
    m6: 30,
    m7: 30,
    m8: 30,
    m9: 29,
    m10: 29,
    m11: 30,
    m12: 31,
    nepYear: 2007
  },
  {
    m1: 31,
    m2: 31,
    m3: 31,
    m4: 32,
    m5: 31,
    m6: 31,
    m7: 29,
    m8: 30,
    m9: 30,
    m10: 29,
    m11: 29,
    m12: 31,
    nepYear: 2008
  },
  {
    m1: 31,
    m2: 31,
    m3: 32,
    m4: 31,
    m5: 31,
    m6: 31,
    m7: 30,
    m8: 29,
    m9: 30,
    m10: 29,
    m11: 30,
    m12: 30,
    nepYear: 2009
  },
  {
    m1: 31,
    m2: 31,
    m3: 32,
    m4: 32,
    m5: 31,
    m6: 30,
    m7: 30,
    m8: 29,
    m9: 30,
    m10: 29,
    m11: 30,
    m12: 30,
    nepYear: 2010
  },
  {
    m1: 31,
    m2: 32,
    m3: 31,
    m4: 32,
    m5: 31,
    m6: 30,
    m7: 30,
    m8: 30,
    m9: 29,
    m10: 29,
    m11: 30,
    m12: 31,
    nepYear: 2011
  },
  {
    m1: 31,
    m2: 31,
    m3: 31,
    m4: 32,
    m5: 31,
    m6: 31,
    m7: 29,
    m8: 30,
    m9: 30,
    m10: 29,
    m11: 30,
    m12: 30,
    nepYear: 2012
  },
  {
    m1: 31,
    m2: 31,
    m3: 32,
    m4: 31,
    m5: 31,
    m6: 31,
    m7: 30,
    m8: 29,
    m9: 30,
    m10: 29,
    m11: 30,
    m12: 30,
    nepYear: 2013
  },
  {
    m1: 31,
    m2: 31,
    m3: 32,
    m4: 32,
    m5: 31,
    m6: 30,
    m7: 30,
    m8: 29,
    m9: 30,
    m10: 29,
    m11: 30,
    m12: 30,
    nepYear: 2014
  },
  {
    m1: 31,
    m2: 32,
    m3: 31,
    m4: 32,
    m5: 31,
    m6: 30,
    m7: 30,
    m8: 30,
    m9: 29,
    m10: 29,
    m11: 30,
    m12: 31,
    nepYear: 2015
  },
  {
    m1: 31,
    m2: 31,
    m3: 31,
    m4: 32,
    m5: 31,
    m6: 31,
    m7: 29,
    m8: 30,
    m9: 30,
    m10: 29,
    m11: 30,
    m12: 30,
    nepYear: 2016
  },
  {
    m1: 31,
    m2: 31,
    m3: 32,
    m4: 31,
    m5: 31,
    m6: 31,
    m7: 30,
    m8: 29,
    m9: 30,
    m10: 29,
    m11: 30,
    m12: 30,
    nepYear: 2017
  },
  {
    m1: 31,
    m2: 32,
    m3: 31,
    m4: 32,
    m5: 31,
    m6: 30,
    m7: 30,
    m8: 29,
    m9: 30,
    m10: 29,
    m11: 30,
    m12: 30,
    nepYear: 2018
  },
  {
    m1: 31,
    m2: 32,
    m3: 31,
    m4: 32,
    m5: 31,
    m6: 30,
    m7: 30,
    m8: 30,
    m9: 29,
    m10: 30,
    m11: 29,
    m12: 31,
    nepYear: 2019
  },
  {
    m1: 31,
    m2: 31,
    m3: 31,
    m4: 32,
    m5: 31,
    m6: 31,
    m7: 30,
    m8: 29,
    m9: 30,
    m10: 29,
    m11: 30,
    m12: 30,
    nepYear: 2020
  },
  {
    m1: 31,
    m2: 31,
    m3: 32,
    m4: 31,
    m5: 31,
    m6: 31,
    m7: 30,
    m8: 29,
    m9: 30,
    m10: 29,
    m11: 30,
    m12: 30,
    nepYear: 2021
  },
  {
    m1: 31,
    m2: 32,
    m3: 31,
    m4: 32,
    m5: 31,
    m6: 30,
    m7: 30,
    m8: 30,
    m9: 29,
    m10: 29,
    m11: 30,
    m12: 30,
    nepYear: 2022
  },
  {
    m1: 31,
    m2: 32,
    m3: 31,
    m4: 32,
    m5: 31,
    m6: 30,
    m7: 30,
    m8: 30,
    m9: 29,
    m10: 30,
    m11: 29,
    m12: 31,
    nepYear: 2023
  },
  {
    m1: 31,
    m2: 31,
    m3: 31,
    m4: 32,
    m5: 31,
    m6: 31,
    m7: 30,
    m8: 29,
    m9: 30,
    m10: 29,
    m11: 30,
    m12: 30,
    nepYear: 2024
  },
  {
    m1: 31,
    m2: 31,
    m3: 32,
    m4: 31,
    m5: 31,
    m6: 31,
    m7: 30,
    m8: 29,
    m9: 30,
    m10: 29,
    m11: 30,
    m12: 30,
    nepYear: 2025
  },
  {
    m1: 31,
    m2: 32,
    m3: 31,
    m4: 32,
    m5: 31,
    m6: 30,
    m7: 30,
    m8: 30,
    m9: 29,
    m10: 29,
    m11: 30,
    m12: 31,
    nepYear: 2026
  },
  {
    m1: 30,
    m2: 32,
    m3: 31,
    m4: 32,
    m5: 31,
    m6: 30,
    m7: 30,
    m8: 30,
    m9: 29,
    m10: 30,
    m11: 29,
    m12: 31,
    nepYear: 2027
  },
  {
    m1: 31,
    m2: 31,
    m3: 32,
    m4: 31,
    m5: 31,
    m6: 31,
    m7: 30,
    m8: 29,
    m9: 30,
    m10: 29,
    m11: 30,
    m12: 30,
    nepYear: 2028
  },
  {
    m1: 31,
    m2: 31,
    m3: 32,
    m4: 31,
    m5: 32,
    m6: 30,
    m7: 30,
    m8: 29,
    m9: 30,
    m10: 29,
    m11: 30,
    m12: 30,
    nepYear: 2029
  },
  {
    m1: 31,
    m2: 32,
    m3: 31,
    m4: 32,
    m5: 31,
    m6: 30,
    m7: 30,
    m8: 30,
    m9: 29,
    m10: 29,
    m11: 30,
    m12: 31,
    nepYear: 2030
  },
  {
    m1: 30,
    m2: 32,
    m3: 31,
    m4: 32,
    m5: 31,
    m6: 30,
    m7: 30,
    m8: 30,
    m9: 29,
    m10: 30,
    m11: 29,
    m12: 31,
    nepYear: 2031
  },
  {
    m1: 31,
    m2: 31,
    m3: 32,
    m4: 31,
    m5: 31,
    m6: 31,
    m7: 30,
    m8: 29,
    m9: 30,
    m10: 29,
    m11: 30,
    m12: 30,
    nepYear: 2032
  },
  {
    m1: 31,
    m2: 31,
    m3: 32,
    m4: 32,
    m5: 31,
    m6: 30,
    m7: 30,
    m8: 29,
    m9: 30,
    m10: 29,
    m11: 30,
    m12: 30,
    nepYear: 2033
  },
  {
    m1: 31,
    m2: 32,
    m3: 31,
    m4: 32,
    m5: 31,
    m6: 30,
    m7: 30,
    m8: 30,
    m9: 29,
    m10: 29,
    m11: 30,
    m12: 31,
    nepYear: 2034
  },
  {
    m1: 30,
    m2: 32,
    m3: 31,
    m4: 32,
    m5: 31,
    m6: 31,
    m7: 29,
    m8: 30,
    m9: 30,
    m10: 29,
    m11: 29,
    m12: 31,
    nepYear: 2035
  },
  {
    m1: 31,
    m2: 31,
    m3: 32,
    m4: 31,
    m5: 31,
    m6: 31,
    m7: 30,
    m8: 29,
    m9: 30,
    m10: 29,
    m11: 30,
    m12: 30,
    nepYear: 2036
  },
  {
    m1: 31,
    m2: 31,
    m3: 32,
    m4: 32,
    m5: 31,
    m6: 30,
    m7: 30,
    m8: 29,
    m9: 30,
    m10: 29,
    m11: 30,
    m12: 30,
    nepYear: 2037
  },
  {
    m1: 31,
    m2: 32,
    m3: 31,
    m4: 32,
    m5: 31,
    m6: 30,
    m7: 30,
    m8: 30,
    m9: 29,
    m10: 29,
    m11: 30,
    m12: 31,
    nepYear: 2038
  },
  {
    m1: 31,
    m2: 31,
    m3: 31,
    m4: 32,
    m5: 31,
    m6: 31,
    m7: 29,
    m8: 30,
    m9: 30,
    m10: 29,
    m11: 30,
    m12: 30,
    nepYear: 2039
  },
  {
    m1: 31,
    m2: 31,
    m3: 32,
    m4: 31,
    m5: 31,
    m6: 31,
    m7: 30,
    m8: 29,
    m9: 30,
    m10: 29,
    m11: 30,
    m12: 30,
    nepYear: 2040
  },
  {
    m1: 31,
    m2: 31,
    m3: 32,
    m4: 32,
    m5: 31,
    m6: 30,
    m7: 30,
    m8: 29,
    m9: 30,
    m10: 29,
    m11: 30,
    m12: 30,
    nepYear: 2041
  },
  {
    m1: 31,
    m2: 32,
    m3: 31,
    m4: 32,
    m5: 31,
    m6: 30,
    m7: 30,
    m8: 30,
    m9: 29,
    m10: 29,
    m11: 30,
    m12: 31,
    nepYear: 2042
  },
  {
    m1: 31,
    m2: 31,
    m3: 31,
    m4: 32,
    m5: 31,
    m6: 31,
    m7: 29,
    m8: 30,
    m9: 30,
    m10: 29,
    m11: 30,
    m12: 30,
    nepYear: 2043
  },
  {
    m1: 31,
    m2: 31,
    m3: 32,
    m4: 31,
    m5: 31,
    m6: 31,
    m7: 30,
    m8: 29,
    m9: 30,
    m10: 29,
    m11: 30,
    m12: 30,
    nepYear: 2044
  },
  {
    m1: 31,
    m2: 32,
    m3: 31,
    m4: 32,
    m5: 31,
    m6: 30,
    m7: 30,
    m8: 29,
    m9: 30,
    m10: 29,
    m11: 30,
    m12: 30,
    nepYear: 2045
  },
  {
    m1: 31,
    m2: 32,
    m3: 31,
    m4: 32,
    m5: 31,
    m6: 30,
    m7: 30,
    m8: 30,
    m9: 29,
    m10: 29,
    m11: 30,
    m12: 31,
    nepYear: 2046
  },
  {
    m1: 31,
    m2: 31,
    m3: 31,
    m4: 32,
    m5: 31,
    m6: 31,
    m7: 30,
    m8: 29,
    m9: 30,
    m10: 29,
    m11: 30,
    m12: 30,
    nepYear: 2047
  },
  {
    m1: 31,
    m2: 31,
    m3: 32,
    m4: 31,
    m5: 31,
    m6: 31,
    m7: 30,
    m8: 29,
    m9: 30,
    m10: 29,
    m11: 30,
    m12: 30,
    nepYear: 2048
  },
  {
    m1: 31,
    m2: 32,
    m3: 31,
    m4: 32,
    m5: 31,
    m6: 30,
    m7: 30,
    m8: 30,
    m9: 29,
    m10: 29,
    m11: 30,
    m12: 30,
    nepYear: 2049
  },
  {
    m1: 31,
    m2: 32,
    m3: 31,
    m4: 32,
    m5: 31,
    m6: 30,
    m7: 30,
    m8: 30,
    m9: 29,
    m10: 30,
    m11: 29,
    m12: 31,
    nepYear: 2050
  },
  {
    m1: 31,
    m2: 31,
    m3: 31,
    m4: 32,
    m5: 31,
    m6: 31,
    m7: 30,
    m8: 29,
    m9: 30,
    m10: 29,
    m11: 30,
    m12: 30,
    nepYear: 2051
  },
  {
    m1: 31,
    m2: 31,
    m3: 32,
    m4: 31,
    m5: 31,
    m6: 31,
    m7: 30,
    m8: 29,
    m9: 30,
    m10: 29,
    m11: 30,
    m12: 30,
    nepYear: 2052
  },
  {
    m1: 31,
    m2: 32,
    m3: 31,
    m4: 32,
    m5: 31,
    m6: 30,
    m7: 30,
    m8: 30,
    m9: 29,
    m10: 29,
    m11: 30,
    m12: 30,
    nepYear: 2053
  },
  {
    m1: 31,
    m2: 32,
    m3: 31,
    m4: 32,
    m5: 31,
    m6: 30,
    m7: 30,
    m8: 30,
    m9: 29,
    m10: 30,
    m11: 29,
    m12: 31,
    nepYear: 2054
  },
  {
    m1: 31,
    m2: 31,
    m3: 32,
    m4: 31,
    m5: 31,
    m6: 31,
    m7: 30,
    m8: 29,
    m9: 30,
    m10: 29,
    m11: 30,
    m12: 30,
    nepYear: 2055
  },
  {
    m1: 31,
    m2: 31,
    m3: 32,
    m4: 31,
    m5: 32,
    m6: 30,
    m7: 30,
    m8: 29,
    m9: 30,
    m10: 29,
    m11: 30,
    m12: 30,
    nepYear: 2056
  },
  {
    m1: 31,
    m2: 32,
    m3: 31,
    m4: 32,
    m5: 31,
    m6: 30,
    m7: 30,
    m8: 30,
    m9: 29,
    m10: 29,
    m11: 30,
    m12: 31,
    nepYear: 2057
  },
  {
    m1: 30,
    m2: 32,
    m3: 31,
    m4: 32,
    m5: 31,
    m6: 30,
    m7: 30,
    m8: 30,
    m9: 29,
    m10: 30,
    m11: 29,
    m12: 31,
    nepYear: 2058
  },
  {
    m1: 31,
    m2: 31,
    m3: 32,
    m4: 31,
    m5: 31,
    m6: 31,
    m7: 30,
    m8: 29,
    m9: 30,
    m10: 29,
    m11: 30,
    m12: 30,
    nepYear: 2059
  },
  {
    m1: 31,
    m2: 31,
    m3: 32,
    m4: 32,
    m5: 31,
    m6: 30,
    m7: 30,
    m8: 29,
    m9: 30,
    m10: 29,
    m11: 30,
    m12: 30,
    nepYear: 2060
  },
  {
    m1: 31,
    m2: 32,
    m3: 31,
    m4: 32,
    m5: 31,
    m6: 30,
    m7: 30,
    m8: 30,
    m9: 29,
    m10: 29,
    m11: 30,
    m12: 31,
    nepYear: 2061
  },
  {
    m1: 30,
    m2: 32,
    m3: 31,
    m4: 32,
    m5: 31,
    m6: 31,
    m7: 29,
    m8: 30,
    m9: 29,
    m10: 30,
    m11: 29,
    m12: 31,
    nepYear: 2062
  },
  {
    m1: 31,
    m2: 31,
    m3: 32,
    m4: 31,
    m5: 31,
    m6: 31,
    m7: 30,
    m8: 29,
    m9: 30,
    m10: 29,
    m11: 30,
    m12: 30,
    nepYear: 2063
  },
  {
    m1: 31,
    m2: 31,
    m3: 32,
    m4: 32,
    m5: 31,
    m6: 30,
    m7: 30,
    m8: 29,
    m9: 30,
    m10: 29,
    m11: 30,
    m12: 30,
    nepYear: 2064
  },
  {
    m1: 31,
    m2: 32,
    m3: 31,
    m4: 32,
    m5: 31,
    m6: 30,
    m7: 30,
    m8: 30,
    m9: 29,
    m10: 29,
    m11: 30,
    m12: 31,
    nepYear: 2065
  },
  {
    m1: 31,
    m2: 31,
    m3: 31,
    m4: 32,
    m5: 31,
    m6: 31,
    m7: 29,
    m8: 30,
    m9: 30,
    m10: 29,
    m11: 29,
    m12: 31,
    nepYear: 2066
  },
  {
    m1: 31,
    m2: 31,
    m3: 32,
    m4: 31,
    m5: 31,
    m6: 31,
    m7: 30,
    m8: 29,
    m9: 30,
    m10: 29,
    m11: 30,
    m12: 30,
    nepYear: 2067
  },
  {
    m1: 31,
    m2: 31,
    m3: 32,
    m4: 32,
    m5: 31,
    m6: 30,
    m7: 30,
    m8: 29,
    m9: 30,
    m10: 29,
    m11: 30,
    m12: 30,
    nepYear: 2068
  },
  {
    m1: 31,
    m2: 32,
    m3: 31,
    m4: 32,
    m5: 31,
    m6: 30,
    m7: 30,
    m8: 30,
    m9: 29,
    m10: 29,
    m11: 30,
    m12: 31,
    nepYear: 2069
  },
  {
    m1: 31,
    m2: 31,
    m3: 31,
    m4: 32,
    m5: 31,
    m6: 31,
    m7: 29,
    m8: 30,
    m9: 30,
    m10: 29,
    m11: 30,
    m12: 30,
    nepYear: 2070
  },
  {
    m1: 31,
    m2: 31,
    m3: 32,
    m4: 31,
    m5: 31,
    m6: 31,
    m7: 30,
    m8: 29,
    m9: 30,
    m10: 29,
    m11: 30,
    m12: 30,
    nepYear: 2071
  },
  {
    m1: 31,
    m2: 32,
    m3: 31,
    m4: 32,
    m5: 31,
    m6: 30,
    m7: 30,
    m8: 29,
    m9: 30,
    m10: 29,
    m11: 30,
    m12: 30,
    nepYear: 2072
  },
  {
    m1: 31,
    m2: 32,
    m3: 31,
    m4: 32,
    m5: 31,
    m6: 30,
    m7: 30,
    m8: 30,
    m9: 29,
    m10: 29,
    m11: 30,
    m12: 31,
    nepYear: 2073
  },
  {
    m1: 31,
    m2: 31,
    m3: 31,
    m4: 32,
    m5: 31,
    m6: 31,
    m7: 30,
    m8: 29,
    m9: 30,
    m10: 29,
    m11: 30,
    m12: 30,
    nepYear: 2074
  },
  {
    m1: 31,
    m2: 31,
    m3: 32,
    m4: 31,
    m5: 31,
    m6: 31,
    m7: 30,
    m8: 29,
    m9: 30,
    m10: 29,
    m11: 30,
    m12: 30,
    nepYear: 2075
  },
  {
    m1: 31,
    m2: 32,
    m3: 31,
    m4: 32,
    m5: 31,
    m6: 30,
    m7: 30,
    m8: 30,
    m9: 29,
    m10: 29,
    m11: 30,
    m12: 30,
    nepYear: 2076
  },
  {
    m1: 31,
    m2: 32,
    m3: 31,
    m4: 32,
    m5: 31,
    m6: 30,
    m7: 30,
    m8: 30,
    m9: 29,
    m10: 30,
    m11: 29,
    m12: 31,
    nepYear: 2077
  },
  {
    m1: 31,
    m2: 31,
    m3: 31,
    m4: 32,
    m5: 31,
    m6: 31,
    m7: 30,
    m8: 29,
    m9: 30,
    m10: 29,
    m11: 30,
    m12: 30,
    nepYear: 2078
  },
  {
    m1: 31,
    m2: 31,
    m3: 32,
    m4: 31,
    m5: 31,
    m6: 31,
    m7: 30,
    m8: 29,
    m9: 30,
    m10: 29,
    m11: 30,
    m12: 30,
    nepYear: 2079
  },
  {
    m1: 31,
    m2: 32,
    m3: 31,
    m4: 32,
    m5: 31,
    m6: 30,
    m7: 30,
    m8: 30,
    m9: 29,
    m10: 29,
    m11: 30,
    m12: 30,
    nepYear: 2080
  },
  {
    m1: 31,
    m2: 32,
    m3: 31,
    m4: 32,
    m5: 31,
    m6: 30,
    m7: 30,
    m8: 30,
    m9: 29,
    m10: 30,
    m11: 29,
    m12: 31,
    nepYear: 2081
  },
  {
    m1: 31,
    m2: 31,
    m3: 32,
    m4: 31,
    m5: 31,
    m6: 31,
    m7: 30,
    m8: 29,
    m9: 30,
    m10: 29,
    m11: 30,
    m12: 30,
    nepYear: 2082
  },
  {
    m1: 31,
    m2: 31,
    m3: 32,
    m4: 31,
    m5: 31,
    m6: 30,
    m7: 30,
    m8: 30,
    m9: 29,
    m10: 30,
    m11: 30,
    m12: 30,
    nepYear: 2083
  },
  {
    m1: 31,
    m2: 31,
    m3: 32,
    m4: 31,
    m5: 31,
    m6: 30,
    m7: 30,
    m8: 30,
    m9: 29,
    m10: 30,
    m11: 30,
    m12: 30,
    nepYear: 2084
  },
  {
    m1: 31,
    m2: 32,
    m3: 31,
    m4: 32,
    m5: 30,
    m6: 31,
    m7: 30,
    m8: 30,
    m9: 29,
    m10: 30,
    m11: 30,
    m12: 30,
    nepYear: 2085
  },
  {
    m1: 30,
    m2: 32,
    m3: 31,
    m4: 32,
    m5: 31,
    m6: 30,
    m7: 30,
    m8: 30,
    m9: 29,
    m10: 30,
    m11: 30,
    m12: 30,
    nepYear: 2086
  },
  {
    m1: 31,
    m2: 31,
    m3: 32,
    m4: 31,
    m5: 31,
    m6: 31,
    m7: 30,
    m8: 30,
    m9: 29,
    m10: 30,
    m11: 30,
    m12: 30,
    nepYear: 2087
  },
  {
    m1: 30,
    m2: 31,
    m3: 32,
    m4: 32,
    m5: 30,
    m6: 31,
    m7: 30,
    m8: 30,
    m9: 29,
    m10: 30,
    m11: 30,
    m12: 30,
    nepYear: 2088
  },
  {
    m1: 30,
    m2: 32,
    m3: 31,
    m4: 32,
    m5: 31,
    m6: 30,
    m7: 30,
    m8: 30,
    m9: 29,
    m10: 30,
    m11: 30,
    m12: 30,
    nepYear: 2089
  },
  {
    m1: 30,
    m2: 32,
    m3: 31,
    m4: 32,
    m5: 31,
    m6: 30,
    m7: 30,
    m8: 30,
    m9: 29,
    m10: 30,
    m11: 30,
    m12: 30,
    nepYear: 2090
  },
  {
    m1: 31,
    m2: 31,
    m3: 32,
    m4: 32,
    m5: 31,
    m6: 30,
    m7: 30,
    m8: 29,
    m9: 30,
    m10: 29,
    m11: 30,
    m12: 30,
    nepYear: 2091
  },
  {
    m1: 31,
    m2: 32,
    m3: 31,
    m4: 32,
    m5: 31,
    m6: 30,
    m7: 30,
    m8: 30,
    m9: 29,
    m10: 29,
    m11: 30,
    m12: 31,
    nepYear: 2092
  },
  {
    m1: 31,
    m2: 31,
    m3: 31,
    m4: 32,
    m5: 31,
    m6: 31,
    m7: 29,
    m8: 30,
    m9: 30,
    m10: 29,
    m11: 29,
    m12: 31,
    nepYear: 2093
  },
  {
    m1: 31,
    m2: 31,
    m3: 32,
    m4: 31,
    m5: 31,
    m6: 31,
    m7: 30,
    m8: 29,
    m9: 30,
    m10: 29,
    m11: 30,
    m12: 30,
    nepYear: 2094
  },
  {
    m1: 31,
    m2: 31,
    m3: 32,
    m4: 32,
    m5: 31,
    m6: 30,
    m7: 30,
    m8: 29,
    m9: 30,
    m10: 29,
    m11: 30,
    m12: 30,
    nepYear: 2095
  },
  {
    m1: 31,
    m2: 32,
    m3: 31,
    m4: 32,
    m5: 31,
    m6: 30,
    m7: 30,
    m8: 30,
    m9: 29,
    m10: 29,
    m11: 30,
    m12: 31,
    nepYear: 2096
  },
  {
    m1: 31,
    m2: 31,
    m3: 31,
    m4: 32,
    m5: 31,
    m6: 31,
    m7: 29,
    m8: 30,
    m9: 30,
    m10: 29,
    m11: 30,
    m12: 30,
    nepYear: 2097
  },
  {
    m1: 31,
    m2: 31,
    m3: 32,
    m4: 31,
    m5: 31,
    m6: 31,
    m7: 30,
    m8: 29,
    m9: 30,
    m10: 29,
    m11: 30,
    m12: 30,
    nepYear: 2098
  },
  {
    m1: 31,
    m2: 31,
    m3: 32,
    m4: 32,
    m5: 31,
    m6: 30,
    m7: 30,
    m8: 29,
    m9: 30,
    m10: 29,
    m11: 30,
    m12: 30,
    nepYear: 2099
  },
  {
    m1: 31,
    m2: 32,
    m3: 31,
    m4: 32,
    m5: 31,
    m6: 30,
    m7: 30,
    m8: 30,
    m9: 29,
    m10: 29,
    m11: 30,
    m12: 31,
    nepYear: 2100
  },
  {
    m1: 31,
    m2: 31,
    m3: 31,
    m4: 32,
    m5: 31,
    m6: 31,
    m7: 29,
    m8: 30,
    m9: 30,
    m10: 29,
    m11: 30,
    m12: 30,
    nepYear: 2101
  },
  {
    m1: 31,
    m2: 31,
    m3: 32,
    m4: 31,
    m5: 31,
    m6: 31,
    m7: 30,
    m8: 29,
    m9: 30,
    m10: 29,
    m11: 30,
    m12: 30,
    nepYear: 2102
  },
  {
    m1: 31,
    m2: 32,
    m3: 31,
    m4: 32,
    m5: 31,
    m6: 30,
    m7: 30,
    m8: 29,
    m9: 30,
    m10: 29,
    m11: 30,
    m12: 30,
    nepYear: 2103
  },
  {
    m1: 31,
    m2: 32,
    m3: 31,
    m4: 32,
    m5: 31,
    m6: 30,
    m7: 30,
    m8: 30,
    m9: 29,
    m10: 30,
    m11: 29,
    m12: 31,
    nepYear: 2104
  },
  {
    m1: 31,
    m2: 31,
    m3: 31,
    m4: 32,
    m5: 31,
    m6: 32,
    m7: 30,
    m8: 29,
    m9: 30,
    m10: 29,
    m11: 30,
    m12: 30,
    nepYear: 2105
  },
  {
    m1: 31,
    m2: 31,
    m3: 32,
    m4: 31,
    m5: 31,
    m6: 31,
    m7: 30,
    m8: 29,
    m9: 30,
    m10: 29,
    m11: 30,
    m12: 30,
    nepYear: 2106
  },
  {
    m1: 31,
    m2: 32,
    m3: 31,
    m4: 32,
    m5: 31,
    m6: 30,
    m7: 30,
    m8: 30,
    m9: 29,
    m10: 29,
    m11: 30,
    m12: 30,
    nepYear: 2107
  },
  {
    m1: 31,
    m2: 32,
    m3: 31,
    m4: 32,
    m5: 31,
    m6: 30,
    m7: 30,
    m8: 30,
    m9: 29,
    m10: 30,
    m11: 29,
    m12: 31,
    nepYear: 2108
  },
  {
    m1: 31,
    m2: 31,
    m3: 32,
    m4: 31,
    m5: 31,
    m6: 31,
    m7: 30,
    m8: 29,
    m9: 30,
    m10: 29,
    m11: 30,
    m12: 30,
    nepYear: 2109
  },
  {
    m1: 31,
    m2: 31,
    m3: 32,
    m4: 31,
    m5: 31,
    m6: 31,
    m7: 30,
    m8: 29,
    m9: 30,
    m10: 29,
    m11: 30,
    m12: 30,
    nepYear: 2110
  },
  {
    m1: 30,
    m2: 31,
    m3: 30,
    m4: 32,
    m5: 30,
    m6: 30,
    m7: 32,
    m8: 30,
    m9: 30,
    m10: 30,
    m11: 30,
    m12: 30,
    nepYear: 2111
  }
];

// src/utilities/get-ad-date-from-bs.ts
var import_dayjs = __toESM(require("dayjs"));

// src/utilities/get-bs-month-total-days.ts
var getBSMonthTotalDays = (bsMonth, bsYear) => {
  const months2 = BikramSambat.getBikramSambatMonths(bsYear);
  const bsMonthIndex = bsMonth - 1;
  return months2.at(bsMonthIndex).numberOfDays;
};

// src/utilities/get-months-with-cumulative-days.ts
var getMonthsWithCumulativeDays = (year) => {
  return BikramSambat.getBikramSambatMonths(year).reduce(
    (accumulatedMonths, currentMonth) => {
      const cumulativeDays = !accumulatedMonths || accumulatedMonths.length === 0 ? currentMonth.numberOfDays : currentMonth.numberOfDays + accumulatedMonths.at(-1).cumulativeDays;
      return [
        ...accumulatedMonths,
        {
          monthNumber: currentMonth.monthNumber,
          monthName: currentMonth.monthName,
          cumulativeDays
        }
      ];
    },
    []
  );
};

// src/utilities/get-ad-date-from-bs.ts
var adDateFromBS = (bsYear, bsMonth, bsDay) => {
  const matchingCalendarPeriod = nepEngCalenderMaps.find(
    (calendarPeriod) => calendarPeriod.nepYear === bsYear
  );
  if (!matchingCalendarPeriod)
    throw new Error(
      `No corresponding data found for the Bikram Sambat year ${bsYear}.`
    );
  const totalPossibleDays = getBSMonthTotalDays(bsMonth, bsYear);
  if (bsDay < 1 || bsDay > totalPossibleDays)
    throw new Error(
      `Invalid date day: ${bsDay}. Day should be between 1 and ${totalPossibleDays} for month ${bsMonth}.`
    );
  const startOfYearDate = new Date(matchingCalendarPeriod.startDate);
  const bsMonthsWithCumulativeDays = getMonthsWithCumulativeDays(bsYear);
  const currentMonthIndex = bsMonth - 1;
  const daysSinceStartOfYear = (bsMonth === 1 ? 0 : bsMonthsWithCumulativeDays[currentMonthIndex - 1].cumulativeDays) + bsDay;
  return (0, import_dayjs.default)(startOfYearDate).add(daysSinceStartOfYear - 1, "days");
};

// src/utilities/is-leap-year.ts
var isLeapYear = (year) => {
  return year % 4 == 0 && year % 100 != 0 || year % 400 == 0;
};

// src/utilities/get-ad-month-total-days.ts
var getADMonthTotalDays = (month, year) => {
  switch (month) {
    case 1:
    case 3:
    case 5:
    case 7:
    case 8:
    case 10:
    case 12:
      return 31;
    case 4:
    case 6:
    case 9:
    case 11:
      return 30;
    case 2:
      return isLeapYear(year) ? 29 : 28;
    default:
      throw new Error(
        `Invalid date month: ${month}. Month should be between 1 and 12.`
      );
  }
};

// src/utilities/parse-ad-string.ts
var parseAdString = (date) => {
  if (typeof date === "string") {
    if (!/^\d{4}-\d{2}-\d{2}$/.test(date))
      throw new Error(
        `Invalid date format: "${date}". Expected format is YYYY-MM-DD.`
      );
    const [year, month, day] = date.split("-").map((num) => parseInt(num));
    if (month < 1 || month > 12)
      throw new Error(
        `Invalid date month: ${month}. Month should be between 1 and 12.`
      );
    const totalPossibleDays = getADMonthTotalDays(month, year);
    if (day < 1 || month > totalPossibleDays)
      throw new Error(
        `Invalid date day: ${day}. Day should be between 1 and ${totalPossibleDays} for month ${month}.`
      );
    return new Date(date);
  }
  throw new Error("Unable to parse AD string.");
};

// src/utilities/get-bs-data-isbetween.ts
var getBoundaryInclusion = (value) => {
  switch (value) {
    case "()":
      return [1, 1];
    case "(]":
      return [1, 0];
    case "[)":
      return [0, 1];
    case "[]":
      return [0, 0];
    default:
      throw new Error(`Invalid boundary inclusion: ${value}`);
  }
};

// src/index.ts
var MIN_BS_YEAR = 1970;
var MAX_BS_YEAR = 2111;
var MIN_MONTH = 1;
var MAX_MONTH = 12;
var BikramSambat = class _BikramSambat {
  /**
   * @deprecated This field will be removed or made private in upcoming versions. Please use `.get('year')` instead.
   */
  bsYear;
  /**
   * @deprecated This field will be removed or made private in upcoming versions. Please use `.get('month')` instead.
   */
  bsMonth;
  /**
   * @deprecated This field will be removed or made private in upcoming versions. Please use `.get('date')` instead.
   */
  bsDay;
  /**
   * @deprecated This field will be removed or made private in upcoming versions. Please use `.get('day')` instead.
   */
  weekDay;
  adDate;
  bsMonthName;
  /**
   * @param {BikramSambatProps} props - The properties of the Bikram Sambat date.
   */
  constructor(props) {
    Object.assign(this, props);
  }
  /**
   * Parses and validates the given BS date and returns an instance of BikramSambat for that date.
   *
   * - BS Year must be between 1970 and 2111
   * - BS Month must be between 1 and 12
   * @param {string} date - BS Date string to be parsed. Valid pattern: YYYY-MM-DD
   * @returns {BikramSambat} An instance of {@link BikramSambat}
   */
  static parse(date) {
    if (!(typeof date === "string"))
      throw new Error("This function only accepts string.");
    if (!/^\d{4}-\d{2}-\d{2}$/.test(date))
      throw new Error(
        `Invalid date format: "${date}". Expected format is YYYY-MM-DD.`
      );
    const [year, month, day] = date.split("-").map((num) => parseInt(num));
    if (year < MIN_BS_YEAR || year > MAX_BS_YEAR)
      throw new Error(
        `Year should be between ${MIN_BS_YEAR} and ${MAX_BS_YEAR}`
      );
    if (month < MIN_MONTH || month > MAX_MONTH)
      throw new Error(
        `Invalid date month: ${month}. Month should be between ${MIN_MONTH} and ${MAX_MONTH}.`
      );
    const currentMonthIndex = month - 1;
    const englishDate = adDateFromBS(year, month, day);
    const bikramSambatDate = {
      weekDay: englishDate.day(),
      bsYear: year,
      bsMonth: month,
      bsDay: day,
      bsMonthName: months[currentMonthIndex].nepName,
      adDate: englishDate.toDate()
    };
    return new _BikramSambat(bikramSambatDate);
  }
  /**
   * Parses and validates the given AD date and returns an instance of BikramSambat for that date.
   * @param {Date | string} date - AD Date object or string to be parsed. Valid pattern for string: YYYY-MM-DD
   * @returns {BikramSambat} An instance of {@link BikramSambat}
   */
  static fromAD(date) {
    if (typeof date === "string") date = parseAdString(date);
    else if (!(date instanceof Date))
      throw new Error("This function only accepts string or Date object.");
    const englishDateDayjs = (0, import_dayjs2.default)(date);
    const matchingCalendarPeriod = nepEngCalenderMaps.find((calendarPeriod) => {
      const startDate = (0, import_dayjs2.default)(calendarPeriod.startDate);
      const endDate = (0, import_dayjs2.default)(calendarPeriod.endDate);
      return endDate.isAfter(englishDateDayjs, "date") && startDate.isBefore(englishDateDayjs, "date") || endDate.isSame(englishDateDayjs, "date") || startDate.isSame(englishDateDayjs, "date");
    });
    if (!matchingCalendarPeriod)
      throw new Error(
        `No corresponding Bikram Sambat date found for the English date ${date.toISOString().split("T")[0]}.`
      );
    const daysSinceStartOfPeriod = englishDateDayjs.diff(matchingCalendarPeriod.startDate, "days") + 1;
    const bsMonthsWithCumulativeDays = getMonthsWithCumulativeDays(
      matchingCalendarPeriod.nepYear
    );
    const currentBsMonth = [...bsMonthsWithCumulativeDays].find(
      (month) => month.cumulativeDays >= daysSinceStartOfPeriod
    );
    if (!currentBsMonth)
      throw new Error(
        `Unable to determine the Bikram Sambat month for the given English date ${date.toISOString().split("T")[0]}.`
      );
    const previousBsMonth = bsMonthsWithCumulativeDays.filter((month) => month.cumulativeDays < daysSinceStartOfPeriod).at(-1);
    const bikramSambatDate = {
      weekDay: englishDateDayjs.day(),
      bsYear: matchingCalendarPeriod.nepYear,
      bsMonth: currentBsMonth.monthNumber,
      bsDay: previousBsMonth ? daysSinceStartOfPeriod - previousBsMonth.cumulativeDays : daysSinceStartOfPeriod,
      bsMonthName: currentBsMonth.monthName,
      adDate: date
    };
    return new _BikramSambat(bikramSambatDate);
  }
  /**
   * Returns a list of months and their total number of days of the year.
   * @param {number} year - Year to get the total number of days for each month.
   * @returns Array of months and their total number of days.
   * @throws {Error} Throws an error if the year is out of range (1970 - 2111) or invalid.
   */
  static getBikramSambatMonths(year) {
    if (year < MIN_BS_YEAR || year > MAX_BS_YEAR)
      throw new Error(
        `Year should be between ${MIN_BS_YEAR} and ${MAX_BS_YEAR}`
      );
    const nepaliYearData = nepDateNoOfDays.find(
      (date) => date.nepYear === year
    );
    if (!nepaliYearData)
      throw new Error(
        `No data available for the year ${year} in the Bikram Sambat calendar.`
      );
    return months.map((month) => ({
      monthNumber: month.id,
      monthName: month.nepName,
      numberOfDays: nepaliYearData[`m${month.id}`]
    }));
  }
  /**
   * Returns the BikramSambat object as a string.
   * @returns {string} Formatted {@link BikramSambat} date as a formatted string `MMMM D YYYY`.
   */
  toString() {
    return `${this.bsMonthName} ${this.bsDay} ${this.bsYear}`;
  }
  /**
   * Returns an instance of BikramSambat for today's date.
   * @returns {BikramSambat} An instance of {@link BikramSambat} for today's date.
   */
  static now() {
    return this.fromAD(/* @__PURE__ */ new Date());
  }
  /**
   * Returns a cloned BikramSambat object with a specified amount of time added.
   * @param {number} value - The amount of the specified unit to add to the date.
   * @param {ManipulateType} [unit='day'] - The unit of time to add. Valid units include 'day', 'month', 'year'.
   * @returns {BikramSambat} A new {@link BikramSambat} date with the specified amount of time added
   */
  add(value, unit = "day") {
    const newDate = {
      bsYear: this.bsYear,
      bsMonth: this.bsMonth,
      bsDay: this.bsDay
    };
    if (unit == "year") {
      newDate.bsYear += value;
      if (newDate.bsYear > MAX_BS_YEAR) {
        throw new Error(
          "The current add operation will exceed the maximum year limit."
        );
      } else if (newDate.bsYear < MIN_BS_YEAR) {
        throw new Error(
          "The current add operation will fall short of the maximum year limit."
        );
      }
    } else if (unit === "month") {
      const totalMonths = newDate.bsMonth + value;
      const yearDelta = Math.floor((totalMonths - 1) / 12);
      newDate.bsMonth = (totalMonths % 12 + 12) % 12 || 12;
      newDate.bsYear += yearDelta;
      if (newDate.bsYear > MAX_BS_YEAR) {
        throw new Error(
          "The current add operation will exceed the maximum year limit."
        );
      } else if (newDate.bsYear < MIN_BS_YEAR) {
        throw new Error(
          "The current add operation will fall short of the minimum year limit."
        );
      }
      if (newDate.bsDay > getBSMonthTotalDays(newDate.bsMonth, newDate.bsYear)) {
        newDate.bsDay = getBSMonthTotalDays(newDate.bsMonth, newDate.bsYear);
      }
    } else {
      return _BikramSambat.fromAD((0, import_dayjs2.default)(this.adDate).add(value, unit).toDate());
    }
    return _BikramSambat.parse(
      `${newDate.bsYear}-${newDate.bsMonth.toString().padStart(2, "0")}-${newDate.bsDay.toString().padStart(2, "0")}`
    );
  }
  /**
   * Returns a cloned BikramSambat object with a specified amount of time subtracted.
   * @param {number} value - The amount of the specified unit to subtract from the date.
   * @param {ManipulateType} [unit='day'] - The unit of time to subtract. Valid units include 'day', 'month', 'year'.
   * @returns {BikramSambat} A new {@link BikramSambat} date with the specified amount of time subtracted
   */
  sub(value, unit = "day") {
    return this.add(value * -1, unit);
  }
  /**
   * Returns a cloned BikramSambat object set to the start of the specified unit.
   * @param {StarOfEndOfType} unit - The unit to set to the start. Valid units include 'day', 'month', 'year'.
   * @returns {BikramSambat} A new {@link BikramSambat} date set to the start of the specified unit
   */
  startOf(unit) {
    const clone = this.clone();
    if (unit === "month") clone.bsDay = 1;
    if (unit === "year") {
      clone.bsDay = 1;
      clone.bsMonth = 1;
      clone.bsMonthName = months.at(0).nepName;
    }
    const newAdDate = adDateFromBS(clone.bsYear, clone.bsMonth, clone.bsDay);
    clone.adDate = newAdDate.toDate();
    clone.weekDay = newAdDate.day();
    return clone;
  }
  /**
   * Returns a cloned BikramSambat object set to the end of the specified unit.
   * @param {StarOfEndOfType} unit - The unit to set to the end. Valid units include 'day', 'month', 'year'.
   * @returns {BikramSambat} A new {@link BikramSambat} date set to the end of the specified unit
   */
  endOf(unit) {
    const clone = this.clone();
    if (unit === "month") {
      clone.bsDay = getBSMonthTotalDays(clone.bsMonth, clone.bsYear);
    } else if (unit === "year") {
      clone.bsMonth = 12;
      clone.bsMonthName = months.at(-1).nepName;
      clone.bsDay = getBSMonthTotalDays(clone.bsMonth, clone.bsYear);
    }
    const newAdDate = adDateFromBS(clone.bsYear, clone.bsMonth, clone.bsDay);
    clone.adDate = newAdDate.toDate();
    clone.weekDay = newAdDate.day();
    return clone;
  }
  /**
   * Formats the BikramSambat date according to the specified format string.
   * Supported tokens:
   * - YYYY: Full year
   * - MM: Month number (zero-padded)
   * - M: Month number
   * - MMMM: Full month name
   * - DD: Date (zero-padded)
   * - D: Date
   * - dddd: Full weekday name
   * - ddd: Abbreviated weekday name
   * - dd: Minimized weekday name
   * - d: Day of the week
   * @param {string} formatString - The format string to use for formatting the date.
   * @returns {string} The formatted date string.
   */
  format(formatString) {
    const REGEX_FORMAT = /Y{4}|M{1,4}|D{1,2}|d{1,4}/g;
    const matches = (match) => {
      switch (match) {
        case "MMMM":
          return this.bsMonthName;
        case "MM":
          return this.bsMonth.toString().padStart(2, "0");
        case "M":
          return this.bsMonth.toString();
        case "YYYY":
          return this.bsYear.toString();
        case "DD":
          return this.bsDay.toString().padStart(2, "0");
        case "D":
          return this.bsDay.toString();
        case "dddd":
          return (0, import_dayjs2.default)(this.adDate).format("dddd");
        case "ddd":
          return (0, import_dayjs2.default)(this.adDate).format("ddd");
        case "dd":
          return (0, import_dayjs2.default)(this.adDate).format("dd");
        case "d":
          return (0, import_dayjs2.default)(this.adDate).format("d");
      }
      return null;
    };
    return formatString.replace(
      REGEX_FORMAT,
      (match) => matches(match) || match
    );
  }
  /**
   * Creates and returns a deep copy of the current BikramSambat instance.
   * @returns {BikramSambat} A new instance of {@link BikramSambat} with the same date properties.
   */
  clone() {
    return new _BikramSambat({
      adDate: new Date(this.adDate),
      bsDay: this.bsDay,
      bsMonth: this.bsMonth,
      bsMonthName: this.bsMonthName,
      bsYear: this.bsYear,
      weekDay: this.weekDay
    });
  }
  /**
   * Checks if the current BikramSambat date is the same as the given date.
   * @param {BikramSambat | Date} date - The date to compare with.
   * @param {UnitType} [unit='day'] - The unit of time to compare. Defaults to 'day'.
   * @returns {boolean} True if the dates are the same, false otherwise.
   * @throws {Error} Throws an error if the value is not a instance of BikramSambat or Date.
   */
  isSame(date, unit = "day") {
    if (unit === "month" || unit === "year") {
      if (date instanceof Date) date = _BikramSambat.fromAD(date);
      if (unit === "month")
        return this.bsMonth === date.bsMonth && this.bsYear === date.bsYear;
      return this.bsYear === date.bsYear;
    }
    if (date instanceof _BikramSambat) date = date.adDate;
    else if (!(date instanceof Date)) throw new Error("Invalid compare value");
    return (0, import_dayjs2.default)(this.adDate).isSame(date, unit);
  }
  /**
   * Checks if the current BikramSambat date is before the given date.
   * @param {BikramSambat | Date} date - The date to compare with.
   * @param {UnitType} [unit='day'] - The unit of time to compare. Defaults to 'day'.
   * @returns {boolean} True if the current date is before the given date, false otherwise.
   * @throws {Error} Throws an error if the value is not a instance of BikramSambat or Date.
   */
  isBefore(date, unit = "day") {
    if (unit === "month" || unit === "year") {
      if (date instanceof Date) date = _BikramSambat.fromAD(date);
      if (unit === "month") {
        if (this.bsYear === date.bsYear) return this.bsMonth < date.bsMonth;
        else if (this.bsYear < date.bsYear) return true;
        return false;
      }
      return this.bsYear < date.bsYear;
    }
    if (date instanceof _BikramSambat) date = date.adDate;
    else if (!(date instanceof Date)) throw new Error("Invalid compare value");
    return (0, import_dayjs2.default)(this.adDate).isBefore(date, unit);
  }
  /**
   * Checks if the current BikramSambat date is after the given date.
   * @param {BikramSambat | Date} date - The date to compare with.
   * @param {UnitType} [unit='day'] - The unit of time to compare. Defaults to 'day'.
   * @returns {boolean} True if the current date is after the given date, false otherwise.
   * @throws {Error} Throws an error if the value is not a instance of BikramSambat or Date.
   */
  isAfter(date, unit = "day") {
    if (unit === "month" || unit === "year") {
      if (date instanceof Date) date = _BikramSambat.fromAD(date);
      if (unit === "month") {
        if (this.bsYear === date.bsYear) return this.bsMonth > date.bsMonth;
        else if (this.bsYear > date.bsYear) return true;
        return false;
      }
      return this.bsYear > date.bsYear;
    }
    if (date instanceof _BikramSambat) date = date.adDate;
    else if (!(date instanceof Date)) throw new Error("Invalid compare value");
    return (0, import_dayjs2.default)(this.adDate).isAfter(date, unit);
  }
  /**
   * Get the value of a specific unit (day, month, year, date).
   *
   * @param {Exclude<UnitType, 'week'>} unit - The unit type to get the value of ('day', 'month', 'year', 'date').
   * @returns {number} The value of the specified unit.
   * @throws {Error} Throws an error if an invalid unit is provided.
   */
  get(unit) {
    switch (unit) {
      case "day":
        return this.weekDay;
      case "month":
        return this.bsMonth;
      case "year":
        return this.bsYear;
      case "date":
        return this.bsDay;
      default:
        throw new Error(`Invalid unit: ${unit}`);
    }
  }
  /**
   * Get or set the Bikram Sambat year.
   *
   * If setting the year, it adjusts the date to ensure it is valid within the new year.
   *
   * @param {number} [value] - The year to set. If not provided, returns the current year.
   * @returns {number | BikramSambat} The current year if no value is provided, otherwise a new BikramSambat instance with the updated year.
   * @throws {Error} Throws an error if the value is not a number or is out of range (1970 - 2111).
   */
  year(value) {
    if (!value) return this.get("year");
    if (typeof value != "number")
      throw new Error(`Invalid value ${value}. Value must be a number.`);
    if (value < MIN_BS_YEAR || value > MAX_BS_YEAR)
      throw new Error(
        `Year should be between ${MIN_BS_YEAR} and ${MAX_BS_YEAR}`
      );
    const currentMonth = this.get("month");
    const totalPossibleDays = getBSMonthTotalDays(currentMonth, value);
    const newDate = this.get("date") > totalPossibleDays ? totalPossibleDays : this.get("date");
    return _BikramSambat.parse(
      `${value}-${currentMonth.toString().padStart(2, "0")}-${newDate.toString().padStart(2, "0")}`
    );
  }
  /**
   * Get or set the Bikram Sambat month.
   *
   * If setting the month, it adjusts the date to ensure it is valid within the new month.
   *
   *
   * @param {number} [value] - The month to set. If not provided, returns the current month.
   * @returns {number|BikramSambat} - The current month if no value is provided, otherwise a new BikramSambat instance with the updated month.
   * @throws {Error} - Throws an error if the value is not a number.
   */
  month(value) {
    if (!value) return this.get("month");
    if (typeof value != "number")
      throw new Error(`Invalid value ${value}. Value must be a number.`);
    value = value - this.get("month");
    const updatedDate = this.add(value, "month");
    const daysInUpdatedMonth = getBSMonthTotalDays(
      updatedDate.get("month"),
      updatedDate.get("year")
    );
    const adjustedDays = daysInUpdatedMonth < this.get("date") ? daysInUpdatedMonth : this.get("date");
    return updatedDate.date(adjustedDays);
  }
  /**
   * Get or set the Bikram Sambat date.
   *
   * @param {number} [value] - The date to set. If not provided, returns the current date.
   * @returns {number | BikramSambat} The current date if no value is provided, otherwise a new BikramSambat instance with the updated date.
   * @throws {Error} Throws an error if the value is not a number.
   */
  date(value) {
    if (!value) return this.get("date");
    if (typeof value != "number")
      throw new Error(`Invalid value ${value}. Value must be a number.`);
    value = value - this.get("date");
    return this.add(value, "day");
  }
  /**
   * Get or set the Bikram Sambat day of the week.
   *
   * @param {number} [value] - The day of the week to set. If not provided, returns the current day of the week.
   * @returns {number|BikramSambat} The current day of the week if no value is provided, otherwise a new BikramSambat instance with the updated day of the week.
   * @throws {Error} Throws an error if the value is not a number.
   */
  day(value) {
    if (!value) return this.get("day");
    if (typeof value != "number")
      throw new Error(`Invalid value ${value}. Value must be a number.`);
    value = value - this.get("day");
    return this.add(value, "day");
  }
  /**
   * Set the value of a specific unit (day, month, year, date).
   *
   * @param {Exclude<UnitType, 'week'>} unit - The unit type to set the value of ('day', 'month', 'year', 'date').
   * @param {number} value - The value to set for the specified unit.
   * @returns {BikramSambat} A new BikramSambat instance with the updated unit value.
   * @throws {Error} Throws an error if the value is not a number.
   */
  set(unit, value) {
    if (typeof value != "number")
      throw new Error(`Invalid value ${value}. Value must be a number.`);
    return this[unit](value);
  }
  /**
   * Determines if a given BikramSambat or Date object falls between two specified dates, considering boundary inclusion.
   *
   * Supported boundaryInclusion values:
   * - '()': Exclude both startDate and endDate
   * - '[]': Include both startDate and endDate
   * - '(]': Exclude startDate, include endDate
   * - '[)': Include startDate, exclude endDate
   *
   * @param {BikramSambat | Date} startDate - The start date to compare against.
   * @param {BikramSambat | Date} endDate - The end date to compare against.
   * @param {ManipulateType} [unit='day'] - The unit of time for comparison. Defaults to 'day'.
   * @param {BoundaryInclusionType} [boundaryInclusion='()'] - The boundary inclusion type for comparison. Defaults to '()'.
   * @returns {boolean} True if the date is between startDate and endDate according to the specified boundaryInclusion, false otherwise.
   */
  isBetween(startDate, endDate, unit = "day", boundaryInclusion = "()") {
    if (startDate instanceof _BikramSambat) startDate = startDate.adDate;
    else if (!(startDate instanceof Date))
      throw new Error("Invalid compare value");
    if (endDate instanceof _BikramSambat) endDate = endDate.adDate;
    else if (!(endDate instanceof Date))
      throw new Error("Invalid compare value");
    const boundaryInclusionpattern = /^[\(\[][\)\]]$/;
    if (!boundaryInclusionpattern.test(boundaryInclusion)) {
      throw new Error("Invalid boundary inclusion pattern");
    }
    const [includeStartDateVal, includeEndDateVal] = getBoundaryInclusion(boundaryInclusion);
    const tempStartDate = (0, import_dayjs2.default)(startDate).subtract(includeStartDateVal, unit);
    const tempEndDate = (0, import_dayjs2.default)(endDate).add(includeEndDateVal, unit);
    if (!(0, import_dayjs2.default)(this.adDate).isAfter(tempStartDate, unit) || !(0, import_dayjs2.default)(this.adDate).isBefore(tempEndDate, unit))
      return false;
    return true;
  }
};
var src_default = BikramSambat;
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  BikramSambat
});
